File: Topic-wise/Subarrays/subarray_sum/subarray_sum.py
"""
    Given an array, print sum of all sub-arrays.
"""

class DSA:
    def subarray_sum(self, arr):
        for i in range(len(arr)):
            for j in range(i, len(arr)):
                print(f"{arr[i:j+1]} - {sum(arr[i:j+1])}")



dsa = DSA()
arr = list(map(int, input().split()))
print(f"Sub-arrays sum of array {arr} are,")
dsa.subarray_sum(arr)

File: Topic-wise/Subarrays/subarray_sum/subarray_sum.rs
/*
    Given an array, print sum of all sub-arrays.
*/

use std::io;

struct DSA;

impl DSA {
    pub fn subarray_sum(arr: Vec<i64>) {
        for i in 0..arr.len() {
            for j in i..arr.len() {
                println!(
                    "{:?} - {}",
                    &arr[i..j + 1],
                    arr[i..j + 1].iter().sum::<i64>()
                );
            }
        }
    }
}

fn main() {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    let arr: Vec<i64> = input
        .trim()
        .split_whitespace()
        .map(|x| x.parse().expect("Not an integer"))
        .collect();

    println!("Sub-arrays sum of array {:?} are,", arr);
    DSA::subarray_sum(arr.clone());
}
